### DISCUSSION TITLE
Unable to reconnect after discord's oauth redirect

### DISCUSSION THREAD

**Discussion by TardisKing40** [pocketbase/pocketbase#1680] [Q&A]
I use cloudflare's tunneling feature and am able to connect to pocketbase after I get redirected.

Before:
![Screenshot_2023-01-25_12-45-56](https://user-images.githubusercontent.com/38845418/214462273-822548de-f52a-4c42-943c-0d12cb920bf7.png)

After:
![Screenshot_2023-01-25_12-48-10](https://user-images.githubusercontent.com/38845418/214462436-2f88b9e0-442b-4fb4-a301-8fcaea5d5f77.png)

**Comment by ganigeorgiev** [comment]
I'm sorry, I don't understand what is the the issue.

Could you elaborate a little more and/or provide a code sample?

If you haven't already, for an OAuth2 web integration you can check the plain js example in https://pocketbase.io/docs/authentication/#web-oauth2-integration.
Also for more detailed errors, you could try starting PocketBase with `--debug`.

**Comment by TardisKing40** [comment]
```
import express from 'express';
import PocketBase from 'pocketbase';
import localStorage from 'localStorage';

const app = express();
const port = 3000;

const pb = new PocketBase("http://pb.tardisking40.me");
const redirect = 'http://app.tardisking40.me/redirect';

app.set('view engine', 'ejs');

// error handling
app.use((err, req, res, next) => {
    console.error(err.stack);
    res.status(500).send('Something broke!');
    res.redirect('/');
    }
);

app.get('/', (req, res) => {
    res.render('index');
    }
);


// find the discord provider in the pocketbase oauth2 providers
app.get('/login', async (req, res) => {
    const authMethods = await pb.collection('users').listAuthMethods();
    try {
        for (const provider of authMethods.authProviders) {
            if (provider.name == 'discord') {
		console.log(provider);
                localStorage.setItem('provider', JSON.stringify(provider));
                res.redirect(provider.authUrl + redirect );
                return;
            }
        }
    } catch (err) {
        console.log(err);
        res.redirect('/');
    }
});


app.get('/redirect', async(req, res) => {
    const { code } = req.query;
    const provider = JSON.parse(localStorage.getItem('provider'));
    console.log(code)
    console.log(provider)
try {
        const authData = await pb.collection('users').authWithOAuth2(
            provider.name,
            code,
            provider.codeVerifier,
            redirect
        )
        // keep the authdata in storage for later use
        localStorage.setItem('authData', JSON.stringify(authData));
        res.redirect('/profile');
    } catch (err) {
        console.log(err);
        res.redirect('/');
    }
});
```

**Reply by ganigeorgiev** [reply]
@TardisKing40

I'm sorry, I still don't understand what is the error and the supposed outcome.

At first glance, it seems that you are processing the OAuth2 redirect server-side (**note that the example in https://pocketbase.io/docs/authentication/#web-oauth2-integration is for the browser!**). I'm not sure what and how `localStorage` in the above works, but if it is shared between all your user clients, then the above implementation is not correct because the "localstorage" need to be isolated for each user. If you want to share client data in a server context between routes you'll have to use some other mechanism (eg. cookie). Additionally, please note that browsers have different behavior when handling redirects and cookies with `SameSite=Strict` policy and you may have to set the cookie attribute to `lax` (for a similar discussion you can check https://github.com/pocketbase/pocketbase/discussions/903#discussioncomment-4369458).

_The OAuth2 integration will be simplified in the near future (v0.14/v0.15) in #55._

**Reply by ganigeorgiev** [reply]
Also, one more thing in relation to the node server-side handling, please note that you must maintain individual SDK instance per each request/client otherwise the `pb.authStore` state will be shared between all your clients and may have unexpected results (aka. you may want to reconsider the global `const pb = new PocketBase("http://pb.tardisking40.me")`). For a SSR example flow and recommendations, you can also check https://github.com/pocketbase/js-sdk#ssr-integration.

**Comment by TardisKing40** [comment]
I am using the cloudflare reverse proxy setting `pb.tardisking40.me` to `publicip:8091`
